sdk/src/test/scala/com/horizen/account/wallet/AccountWalletTest.scala

```diff
@@ -0,0 +1,198 @@
+
+  @Before
+  def setUp() : Unit = {
+
+    // Set base Secrets data
+    secretList ++= getPrivateKey25519List(5).asScala
+    secretVersions += getVersion
+
+    for (s <- secretList) {
+      storedSecretList.append({
+        val key = new ByteArrayWrapper(Blake2b256.hash(s.publicImage().bytes))
+        val value = new ByteArrayWrapper(sidechainSecretsCompanion.toBytes(s))
+        new Pair(key, value)
+      })
+    }
+
+
+    // Mock get and update methods of SecretStorage
+    Mockito.when(mockedSecretStorage.getAll).thenReturn(storedSecretList.asJava)
+
+    Mockito.when(mockedSecretStorage.get(ArgumentMatchers.any[ByteArrayWrapper]()))
+      .thenAnswer(answer => {
+        storedSecretList.filter(_.getKey.equals(answer.getArgument(0)))
+      })
+
+    Mockito.when(mockedSecretStorage.get(ArgumentMatchers.anyList[ByteArrayWrapper]()))
+      .thenAnswer(answer => {
+        storedSecretList.filter(p => answer.getArgument(0).asInstanceOf[JList[ByteArrayWrapper]].contains(p.getKey))
+      })
+
+    Mockito.when(mockedSecretStorage.update(ArgumentMatchers.any[ByteArrayWrapper](),
+        ArgumentMatchers.anyList[Pair[ByteArrayWrapper,ByteArrayWrapper]](),
+        ArgumentMatchers.anyList[ByteArrayWrapper]()))
+      .thenAnswer(answer => {
+        secretVersions.append(answer.getArgument(0))
+        for (s <- answer.getArgument(2).asInstanceOf[JList[ByteArrayWrapper]].asScala) {
+          val index = storedSecretList.indexWhere(p => p.getKey.equals(s))
+          if (index != -1)
+            storedSecretList.remove(index)
+        }
+        for (s <- answer.getArgument(1).asInstanceOf[JList[Pair[ByteArrayWrapper,ByteArrayWrapper]]].asScala) {
+          val index = storedSecretList.indexWhere(p => p.getKey.equals(s.getKey))
+          if (index != -1)
+            storedSecretList.remove(index)
+        }
+        storedSecretList.appendAll(answer.getArgument(1).asInstanceOf[JList[Pair[ByteArrayWrapper,ByteArrayWrapper]]].asScala)
+      })
+
+
+
+
+  }
+
+
+  @Test
+  def testRollback(): Unit = {
+    val mockedSecretStorage: SidechainSecretStorage = mock[SidechainSecretStorage]
+
+    val accountWallet = new AccountWallet(
+      "seed".getBytes(),
+      mockedSecretStorage)
+
+    // Prepare block ID and corresponding version
+    val blockId = new Array[Byte](32)
+    Random.nextBytes(blockId)
+    val versionTag: VersionTag = VersionTag @@ BytesUtils.toHexString(blockId)
+
+    assertTrue("SidechainWallet rollback expected to be successful", accountWallet.rollback(versionTag).isSuccess)
+
+    Mockito.verifyNoInteractions(mockedSecretStorage)
+  }
+
+
+  @Test
+  def testSecrets(): Unit = {
+    val mockedSecretStorage1: SidechainSecretStorage = mock[SidechainSecretStorage]
+    val accountWallet = new AccountWallet(
+      "seed".getBytes(),
+      mockedSecretStorage1)
+    val secret1 = getPrivateKey25519("testSeed1".getBytes())
```

Comment:
The storage is mocked, yet we consider adding also a Secp256k1 `secret3` for tests below